File: corda-training-template/java-source/src/test/java/com/iou/java/transactions/IOUTransactionTests.java
package com.iou.java.transactions;

import com.iou.java.contract.IOUContract;
import com.iou.java.state.IOUState;
import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.crypto.CompositeKey;
import net.corda.core.crypto.Party;
import org.junit.BeforeClass;
import org.junit.Test;

import static net.corda.testing.CoreTestUtils.*;

public class IOUTransactionTests {
    static private final Party miniCorp = getMINI_CORP();
    static private final Party megaCorp = getMEGA_CORP();
    static private final CompositeKey[] keys = new CompositeKey[2];

    @BeforeClass
    public static void setUpClass() {
        keys[0] = getMINI_CORP_PUBKEY();
        keys[1] = getMEGA_CORP_PUBKEY();
    }

//    @Test
//    public void mustIncludeCommand() {
//        ledger(ledgerDSL -> {
//            ledgerDSL.transaction(txDSL -> {
//                txDSL.output(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                txDSL.fails();
//                return null;
//            });
//            return null;
//        });
//    }
//
//    @Test
//    public void cannotCreateNegativeValueIOUs() {
//        ledger(ledgerDSL -> {
//            ledgerDSL.transaction(txDSL -> {
//                txDSL.output(new IOUState(-1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                txDSL.command(keys, IOUContract.Create::new);
//                txDSL.failsWith("The IOU's value must be non-negative.");
//                return null;
//            });
//            return null;
//        });
//    }
//
//    @Test
//    public void createTransactionMustHaveNoInputs() {
//        ledger(ledgerDSL -> {
//            ledgerDSL.transaction(txDSL -> {
//                txDSL.input(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                txDSL.output(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                txDSL.command(keys, IOUContract.Create::new);
//                txDSL.failsWith("No inputs should be consumed when issuing an IOU.");
//                return null;
//            });
//            return null;
//        });
//    }
//
//    @Test
//    public void createTransactionMustHaveOneOutput() {
//        ledger(ledgerDSL -> {
//            ledgerDSL.transaction(txDSL -> {
//                txDSL.output(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                txDSL.output(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                txDSL.command(keys, IOUContract.Create::new);
//                txDSL.failsWith("Only one output state should be created.");
//                return null;
//            });
//            return null;
//        });
//    }
//
//    @Test
//    public void senderMustSignCreateTransaction() {
//        ledger(ledgerDSL -> {
//            ledgerDSL.transaction(txDSL -> {
//                txDSL.output(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                CompositeKey[] keys = new CompositeKey[1];
//                keys[0] = getMEGA_CORP_PUBKEY();
//                txDSL.command(keys, IOUContract.Create::new);
//                txDSL.failsWith("All of the participants must be signers.");
//                return null;
//            });
//            return null;
//        });
//    }
//
//    @Test
//    public void recipientMustSignCreateTransaction() {
//        ledger(ledgerDSL -> {
//            ledgerDSL.transaction(txDSL -> {
//                txDSL.output(new IOUState(1, miniCorp, megaCorp, new IOUContract(), new UniqueIdentifier()));
//                CompositeKey[] keys = new CompositeKey[1];
//                keys[0] = getMINI_CORP_PUBKEY();
//                txDSL.command(keys, IOUContract.Create::new);
//                txDSL.failsWith("All of the participants must be signers.");
//                return null;
//            });
//            return null;
//        });
//    }
}

File: corda-training-template/kotlin-source/src/test/kotlin/com/iou/kotlin/transactions/IOUTransactionTests.kt
package com.iou.kotlin.transactions

import com.iou.kotlin.contract.IOUContract
import com.iou.kotlin.state.IOUState
import net.corda.testing.*
import org.junit.Test

class IOUTransactionTests {

//    @Test
//    fun mustIncludeCommand() {
//        ledger {
//            transaction {
//                output { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                fails()
//            }
//        }
//    }
//
//    @Test
//    fun cannotCreateNegativeValueIOUs() {
//        ledger {
//            transaction {
//                output(IOUState(-1, MINI_CORP, MEGA_CORP, IOUContract()))
//                command(MINI_CORP_PUBKEY, MEGA_CORP_PUBKEY) { IOUContract.Create() }
//                failsWith("The IOU's value must be non-negative.")
//            }
//        }
//    }
//
//    @Test
//    fun createTransactionMustHaveNoInputs() {
//        ledger {
//            transaction {
//                input { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                output { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                command(MINI_CORP_PUBKEY, MEGA_CORP_PUBKEY) { IOUContract.Create() }
//                failsWith("No inputs should be consumed when issuing an IOU.")
//            }
//        }
//    }
//
//    @Test
//    fun createTransactionMustHaveOneOutput() {
//        ledger {
//            transaction {
//                output { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                output { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                command(MINI_CORP_PUBKEY, MEGA_CORP_PUBKEY) { IOUContract.Create() }
//                failsWith("Only one output state should be created.")
//            }
//        }
//    }
//
//    @Test
//    fun senderMustSignCreateTransaction() {
//        ledger {
//            transaction {
//                output { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                command(MEGA_CORP_PUBKEY) { IOUContract.Create() }
//                failsWith("All of the participants must be signers.")
//            }
//        }
//    }
//
//    @Test
//    fun recipientMustSignCreateTransaction() {
//        ledger {
//            transaction {
//                output { IOUState(1, MINI_CORP, MEGA_CORP, IOUContract()) }
//                command(MINI_CORP_PUBKEY) { IOUContract.Create() }
//                failsWith("All of the participants must be signers.")
//            }
//        }
//    }
}